docs: `defineNuxtPlugin` function documentation (#32328)
nuxt/nuxt: docs/3.api/3.utils/define-nuxt-plugin.md

  
    A simple function that receives the Nuxt app instance and can optionally return providers.

  **Object Plugin Properties**

  **`name`**
  - **Type**: `string`
  
    Optional name for the plugin, useful for debugging and dependency management.

  **`enforce`**
  - **Type**: `'pre'` | `'default'` | `'post'`
  
    Controls when the plugin runs relative to other plugins.

  **`dependsOn`**
  - **Type**: `string[]`
  
    Array of plugin names this plugin depends on. Ensures proper execution order.

  **`order`**
  - **Type**: `number`
  
    This allows more granular control over plugin order and should only be used by advanced users. **It overrides the value of `enforce` and is used to sort plugins.**

  **`parallel`**
  - **Type**: `boolean`
  
    Whether to execute the plugin in parallel with other parallel plugins.

  **`setup`**
  - **Type**: `Plugin<T>`
  
    The main plugin function, equivalent to a function plugin.

  **`hooks`**
  - **Type**: `Partial<RuntimeNuxtHooks>`
  
    Nuxt app runtime hooks to register directly.

  **`env`**
  - **Type**: `{ islands?: boolean }`
  

>     Environment configuration. Set this value to `false` if you don't want the plugin to run when rendering server-only or island components.

Reviewer: ```suggestion
    Environment configuration. Set a value to `false` to prevent the plugin running in specified environment.
```

If I remember correctly, setting `env` to `false` doesn't do anything

Author: Yes you right, I made the change!

## Examples

### Basic Usage

The example below demonstrates a simple plugin that adds global functionality:

```ts [plugins/hello.ts]
export default defineNuxtPlugin((nuxtApp) => {
  // Add a global method
  return {
    provide: {
      hello: (name: string) => `Hello ${name}!`
    }
  }
})
```

### Object Syntax Plugin

The example below shows the object syntax with advanced configuration:

```ts [plugins/advanced.ts]
export default defineNuxtPlugin({
  name: 'my-plugin',
  enforce: 'pre',
  async setup (nuxtApp) {
    // Plugin setup logic
    const data = await $fetch('/api/config')
    
    return {
      provide: {
        config: data
      }
    }
  },
  hooks: {
    'app:created'() {
      console.log('App created!')
    }
  },

>   env: {

Reviewer: This may be not necessary (or set it to false) as `true` is the value by default

Author: Done ✅
